File: AutoVoiceServer/contracts/src/test/java/com/autovoice/server/contracts/ContractSmokeTest.java
package com.autovoice.server.contracts;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.net.URISyntaxException;
import java.net.URL;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;

class ContractSmokeTest {

    @Test
    void sharedFixturesReadable() throws IOException, URISyntaxException {
        URL url = ContractSmokeTest.class.getClassLoader().getResource("gateway-reply-action.json");
        Assertions.assertNotNull(url, "shared/fixtures 未接线");
        String text = new String(Files.readAllBytes(Path.of(url.toURI())), StandardCharsets.UTF_8);
        Assertions.assertTrue(text.contains("\"type\": \"reply\""));
    }
}

File: AutoVoice/voice-core/src/test/kotlin/com/autovoice/voicecore/ContractSmokeTest.kt
package com.autovoice.voicecore

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class ContractSmokeTest {
    @Test
    fun `shared fixtures readable`() {
        val path = javaClass.classLoader.getResource("gateway-reply-action.json")
        assertNotNull(path, "shared/fixtures 未接线")
        val text = path!!.readText()
        assertTrue(text.contains("\"type\": \"reply\""))
    }
}
